pkg/server/server_controller_orchestration.go

```diff
@@ -236,6 +255,23 @@ func (c *serverController) startControlledServer(
 		return nil, err
 	}
 
+	var singleShotErrCh chan error
+	if singleShot {
+		// Singleshot means we are not retrying the startup, and the
+		// caller is interested in the startup error if any.
+		singleShotErrCh = make(chan error, 1)
+		defer func() {
+			select {
+			case startupErr := <-singleShotErrCh:
+				resErr = errors.CombineErrors(resErr, startupErr)
+			case <-c.stopper.ShouldQuiesce():
+				resErr = errors.CombineErrors(resErr, errors.New("server shutting down"))
+			case <-topCtx.Done():
+				resErr = errors.CombineErrors(resErr, topCtx.Err())
+			}
+		}()
+	}
```

Comment:
Github won't let me attach the comment to the correct line, but do we need to close the singleShotErrCh in the case of a non-nil error return from RunAsyncTask?